### PR TITLE
Remove toolchain submodule

### PR DESCRIPTION
**Pull request by xross**
* Remove use of submodules to avoid any user confusion 
    * Jenkinsfile manually clones out the toolchain repo to check the legacy build 
 * Add and use xcore_math.h (matching every other lib..) - original xmath/xmath.h remains for backward compatibility 
 * Renamed examples to follow standard structure 
 
 
 (see #193 first)

### CODE REVIEW

**Comment by xross** [comment]
For some reason PRs occasionally fair with "unknown target XK-EVK-XU316" in the cmake step

**Comment by danielpieczko** [comment]
> For some reason PRs occasionally fail with "unknown target XK-EVK-XU316" in the cmake step when building examples

Something weird is happening. Look at this log:
http://srv-bri-jcim0:8080/job/XMOS/job/lib_xcore_math/job/PR-194/36/pipeline-console/log?nodeId=124
A sensible command is run for XCommon CMake, but you don't see the usual output showing the toolchain, etc.

Compare with this from Linux:
http://srv-bri-jcim0:8080/job/XMOS/job/lib_xcore_math/job/PR-194/36/pipeline-console/log?nodeId=134

Makes me wonder if the tools environment isn't loaded, but the Jenkinsfile looks correct using withTools, so not sure what's going on here.

File: Jenkinsfile [hunk 1]
```diff
@@ -115,17 +114,16 @@ pipeline {
             stage('Legacy build') {
               steps {
                 runningOn(env.NODE_NAME)
+                sh 'git clone git@github.com:xmos/xmos_cmake_toolchain'
```
Comment (danielpieczko) [inline Jenkinsfile:117]: Low risk, but could be worth using a tag on the clone here and line 192.

**Review by danielpieczko — APPROVED** [review]
Looks good - I went through this yesterday but was waiting to see what happened with the intermittent Windows build failures, but it looks like those have disappeared.

File: Jenkinsfile (hunk 1 above)
Comment (xross) [reply Jenkinsfile]: I'll clone out v1.0.0

**Comment by xross** [comment]
Build fail is now a website glitch effecting doc build link check ..

**Pull request merged**